tests/iprep-12-rule-types/test.yaml

```diff
@@ -0,0 +1,44 @@
+requires:
+  min-version: 7
+  pcap: false
+
+args:
+- --engine-analysis
+
+checks:
+  - filter:
+      filename: rules.json
+      count: 1
+      match:
+        id: 1
+        type: "ip_only"
+  - filter:
+      filename: rules.json
+      count: 1
+      match:
+        id: 2
+        type: "ip_only"
+  - filter:
+      filename: rules.json
+      count: 1
+      match:
+        id: 3
+        type: "ip_only"
+  - filter:
```

Comment:
Could we do with 2 rules instead of 6 ?

Reply:
possibly there are likely duplicates here and there, but in many cases what I'm doing is trying to check if variations here and there affect the rule type (or state) in some way, so the volume will likely be inevitable.